kbase/narrative
# DATAUP-292 import cell skeleton

# Description of PR purpose/changes

This introduces a skeletal bulk import cell.
Most of what's here is (sort of) boilerplate. There's two parts to it.

First is the new bulk import cell extension. This is under nbextensions/bulkImportCell, with the primary file `main.js`. `main.js` provides the Jupyter requirement, `load_ipython_extension` (see [here](https://jupyter-notebook.readthedocs.io/en/latest/extending/frontend_extensions.html#the-structure-of-a-front-end-extension) for details). In this case, that function does two things.
1. It iterates over the cells in the existing notebook for what would be identified as bulk import cells based on their metadata.
2. It binds an event that turns newly created cells into bulk import cells based on their metadata.
The Bulk Import Cell itself is a basic class that initializes itself (i.e. sets up the metadata that lets the Narrative know that it's a bulk import cell, only run once per created cell), and sets itself up for running (run each time the page gets loaded). It also handles a few tricky things like setting up icons, and setting up a bus for event listening.

The second part is the Narrative tweaks to create a new bulk import cell. I added a function to the main Narrative module to create one, and bound it to the Import Selected button in the staging area.

None of this is perfect, but it's a place to start.

# Jira Ticket / Issue #
e.g. https://kbase-jira.atlassian.net/browse/DATAUP-X

# Testing Instructions
* Details for how to test the PR: 
1. install the narrative again (or run the `nbextensions/install.sh` script) to install the new extension
2. open a local narrative
3. go to the staging area, select a file, and click "Import Selected"

- [ ] Tests pass in travis and locally (tests are incomplete, but it's a start, and I want this opened up for discussion)

# Dev Checklist:


# Updating Version and Release Notes (if applicable)
